File: Chapter13/Java/TestHandlers/app/src/main/java/com/androidbook/handlers/ReportStatusHandler.java
package com.androidbook.handlers;

import android.os.Handler;
import android.os.Message;
import android.util.Log;

import androidx.annotation.NonNull;

public class ReportStatusHandler extends Handler {

    private static final String TAG = "TestHandler2";

    private final TestHandlersDriverActivity parentActivity;

    public ReportStatusHandler(TestHandlersDriverActivity inParentActivity) {
        parentActivity = inParentActivity;
    }

    @Override
    public void handleMessage(@NonNull Message msg) {
        String pm = Utils.getStringFromABundle(msg.getData());

        Log.d(TAG, pm);
        printMessage(pm);
    }

    private void printMessage(String xyz) {
        parentActivity.appendText(xyz);
    }

}

File: Chapter13/Kotlin/TestHandlers/app/src/main/java/com/androidbook/handlers/ReportStatusHandler.kt
package com.androidbook.handlers

import android.os.Handler
import android.os.Looper
import android.os.Message
import android.util.Log

class ReportStatusHandler(
    private val parentActivity: TestHandlersDriverActivity,
    loop: Looper,
) : Handler(loop) {

    override fun handleMessage(msg: Message) {
        val pm = Utils.getStringFromABundle(msg.data)

        Log.d(TAG, pm ?: "null")
        printMessage(pm ?: "null")
    }

    private fun printMessage(xyz: String) {
        parentActivity.appendText(xyz)
    }

    companion object {
        const val TAG = "TestHandler2"
    }
}